### PR TITLE
[trainer] Sanitize both identifier and test case to allow spaces in test name

### PR DESCRIPTION
**Pull request by tahirmt**
Dupe of https://github.com/fastlane/fastlane/pull/21432 so while upstream is updated we can unblock ourselves

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [ ] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Description
If the test case name has spaces or any special characters, the reported `number_of_failures` will always be 0. This will also not include any failed tests inside the junit report.

The reason this was happening was because during the comparison to check the test failure there was a sanitization step that's not applied to the identifier itself so it never matched.

This is from the log output of such a test.
```
** TEST FAILED **
[15:11:03]: Exit status: 65
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 3 |
| Number of failures | 0 |
+--------------------+---+

[15:11:04]: Test execution failed. Exit status: 65
```

### Testing Steps

The easiest way to reproduce it was to use https://github.com/Quick/quick because Quick 7 produces test case names with spaces if the spec names had spaces in it. And when you run `scan` the output will always say the number of failures is 0.

I have attached a sample project.

1. Download project
2. Run `bundle install`
3. Run `bundle exec fastlane test` 

When it runs the output will show number of failures as 0.

Then you can uncomment the branch line in the `Gemfile` and run bundle install and switch to this branch. Then you will see that the number of failure is 1.

[SpaceTests.zip](https://github.com/fastlane/fastlane/files/12221320/SpaceTests.zip)

### CODE REVIEW

File: trainer/spec/test_parser_spec.rb [hunk 1]
```diff
@@ -198,6 +198,44 @@
...
+      it "works as expected with xcresult with spaces", requires_xcode: true do
+        tp = Trainer::TestParser.new("./trainer/spec/fixtures/Test.with_spaces.xcresult")
+        expect(tp.data).to eq([
+                                {
+                                  project_path: "SpaceTests.xcodeproj",
+                                  target_name: "SpaceTestsTests",
+                                  test_name: "SpaceTestsTests",
+                                  configuration_name: "Test Scheme Action",
+                                  duration: 0.21180307865142822,
+                                  tests: [
```
Comment (rayray) [inline trainer/spec/test_parser_spec.rb:211]: Should we also include an xcresult with only passing tests with spaces in their names?

Comment (rayray) [reply trainer/spec/test_parser_spec.rb:211]: I'd also suggest adding a passing test to the xcresult we expect to fail.

Comment (tahirmt) [reply trainer/spec/test_parser_spec.rb:211]: This has 2 passing and one failing test but I can do that

Comment (tahirmt) [reply trainer/spec/test_parser_spec.rb:211]: It's just hard to create the fixture. I'll see what I can do

Comment (tahirmt) [reply trainer/spec/test_parser_spec.rb:211]: @rayray done. Added additional tests

**Review by bmulligan-thescore — APPROVED** [review]

**Review by rayray — APPROVED** [review]

**Pull request merged**